app/models/chargeHistory/AdjustmentHistoryModel.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.chargeHistory
+
+import java.time.LocalDate
+
+case class AdjustmentHistoryModel(creationEvent: AdjustmentModel,
```

Comment:
Looks like we can use NonEmptyList with head representing creation event and history as a tail;
https://typelevel.org/cats/datatypes/nel.html;
but its really up to you to decided if you keen to use this type, current modelling approach is also ok;

Reply:
I think I prefer the current implementation, since we want to handle things differently based on whether the adjustments list is empty, I think this is clearer